src/v1/asset_service_client.ts

```diff
@@ -0,0 +1,1053 @@
+   *     app is running in an environment which supports
+   *     {@link https://developers.google.com/identity/protocols/application-default-credentials Application Default Credentials},
+   *     your project ID will be detected automatically.
+   * @param {function} [options.promise] - Custom promise module to use instead
+   *     of native Promises.
+   * @param {string} [options.apiEndpoint] - The domain name of the
+   *     API remote host.
+   */
+
+  constructor(opts?: ClientOptions) {
+    // Ensure that options include the service address and port.
+    const staticMembers = this.constructor as typeof AssetServiceClient;
+    const servicePath =
+      opts && opts.servicePath
+        ? opts.servicePath
+        : opts && opts.apiEndpoint
+        ? opts.apiEndpoint
+        : staticMembers.servicePath;
+    const port = opts && opts.port ? opts.port : staticMembers.port;
+
+    if (!opts) {
+      opts = {servicePath, port};
+    }
+    opts.servicePath = opts.servicePath || servicePath;
+    opts.port = opts.port || port;
+    opts.clientConfig = opts.clientConfig || {};
+
+    const isBrowser = typeof window !== 'undefined';
+    if (isBrowser) {
+      opts.fallback = true;
+    }
+    // If we are in browser, we are already using fallback because of the
+    // "browser" field in package.json.
+    // But if we were explicitly requested to use fallback, let's do it now.
+    const gaxModule = !isBrowser && opts.fallback ? gax.fallback : gax;
+
+    // Create a `gaxGrpc` object, with any grpc-specific options
+    // sent to the client.
+    opts.scopes = (this.constructor as typeof AssetServiceClient).scopes;
+    const gaxGrpc = new gaxModule.GrpcClient(opts);
+
+    // Save the auth object to the client, for use by other methods.
+    this.auth = gaxGrpc.auth as gax.GoogleAuth;
+
+    // Determine the client header string.
+    const clientHeader = [`gax/${gaxModule.version}`, `gapic/${version}`];
+    if (typeof process !== 'undefined' && 'versions' in process) {
+      clientHeader.push(`gl-node/${process.versions.node}`);
+    } else {
+      clientHeader.push(`gl-web/${gaxModule.version}`);
+    }
+    if (!opts.fallback) {
+      clientHeader.push(`grpc/${gaxGrpc.grpcVersion}`);
+    }
+    if (opts.libName && opts.libVersion) {
+      clientHeader.push(`${opts.libName}/${opts.libVersion}`);
+    }
+    // Load the applicable protos.
+    // For Node.js, pass the path to JSON proto file.
+    // For browsers, pass the JSON content.
+
+    const nodejsProtoPath = path.join(
+      __dirname,
+      '..',
+      '..',
+      'protos',
+      'protos.json'
+    );
+    const protos = gaxGrpc.loadProto(
+      opts.fallback ? require('../../protos/protos.json') : nodejsProtoPath
+    );
+
+    // This API contains "path templates"; forward-slash-separated
+    // identifiers to uniquely identify resources within the API.
+    // Create useful helper objects for these.
+    this._pathTemplates = {
```

Comment:
The list of path templates is different (if you compare it with the deleted `src/v1/asset_service_client.js`).  Let me double check if the new path templates are correct in the annotations. For now, please mark this PR as `feat!` because it now has breaking changes.